nixos/modules/services/networking/pppd.nix

```diff
@@ -0,0 +1,137 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  meta = {
+    maintainers = with maintainers; [ danderson ];
+  };
+
+  options = {
+    services.pppd = {
+      package = mkOption {
+        default = pkgs.ppp;
+        defaultText = "pkgs.ppp";
+        type = types.package;
+        description = "pppd package to use.";
+      };
+
+      peers = mkOption {
+        default = {};
+        type = types.loaOf (types.submodule (
+          { name, ... }:
+          {
+            options = {
+              name = mkOption {
+                example = "dialup";
+                type = types.str;
+                description = "Name of the PPP peer.";
+              };
+
+              enable = mkOption {
```

Comment:
You could use `mkEnableOption` here

Reply:
Ah you want it to be on by default, this is probably fine

Reply:
Yes, in this case I think the default should be true because it only matters if the submodule exists. So the most likely use case for this option is "I want to temporarily disable this 1 peer".